dace/transformation/passes/simplification/control_flow_raising.py

```diff
@@ -139,10 +150,33 @@ def _lift_conditionals(self, sdfg: SDFG) -> int:
                     graph.add_edge(block, conditional, InterstateEdge())
 
                     # Populate branches.
+                    full_cond_expression: Optional[sympy.Basic] = None
+                    uncond_generated = False
                     for i, oe in enumerate(oedges):
                         branch_name = 'branch_' + str(i) + '_' + block.label
                         branch = ControlFlowRegion(branch_name, sdfg)
-                        conditional.add_branch(oe.data.condition, branch)
+
+                        if not oe.data.is_unconditional():
+                            if i == len(oedges) - 1 and oe.data.condition_sympy() == sympy.Not(full_cond_expression):
```

Comment:
You can use ast_negate here and the original condition. We do this in other places where `condition_sympy` is not always possible

Reply:
But the purpose here (and with the comment below) is to evaluate the conditions symbolically to check for symbolic equivalence - won't that be impossible (or I suppose significantly harder) if using Python AST objects instead?